index.js

```diff
@@ -13,7 +13,19 @@ function BroccoliMergeTrees(inputNodes, options) {
     throw new Error('Expected array, got ' + inputNodes)
   }
   options = options || {}
-  Plugin.call(this, inputNodes, {
+  var newInputNodes = []
+  for (var i = 0 ; i < inputNodes.length ; i++ ) {
+    if (inputNodes[i].constructor.name  === this.constructor.name ) {
```

Comment:
we can brand the constructor with a non-guessable symbol (that encodes some version compat stuff) to prevent future issues